nixos/modules/services/networking/cloudflared.nix

```diff
@@ -0,0 +1,95 @@
+{ config, lib, pkgs, ... }:
+
+with lib;
+
+let cfg = config.services.cloudflared;
+in
+{
+  meta.maintainers = with maintainers; [ bbigras ];
+
+  options.services.cloudflared = {
+    enable = mkEnableOption "Cloudflared Argo Tunnel client daemon";
+
+    package = mkOption {
+      type = types.package;
+      default = pkgs.cloudflared;
+      defaultText = "pkgs.cloudflared";
+      description = "The package to use for Cloudflared";
+    };
+
+    tunnels = mkOption {
+      description = ''
+        Cloudflare tunnels
+      '';
+      type = types.attrsOf (types.submodule ({ name, ... }: {
+        options = {
+          tunnel = mkOption {
+            type = with types; nullOr str;
+            default = null;
+            description = ''
+              Tunnel ID
+            '';
+            example = "00000000-0000-0000-0000-000000000000";
+          };
+
+          credentialsFile = mkOption {
+            type = with types; nullOr str;
+            default = null;
+            description = ''
+              credential file
+            '';
+          };
+
+          default = mkOption {
+            type = with types; nullOr str;
+            default = null;
+            description = ''
+              Catch-all if no ingress matches
+            '';
+            example = "http_status:404";
+          };
+
+          ingress = mkOption {
+            type = types.attrsOf types.str;
+            default = { };
+            example = {
+              "*.domain.com" = "http://localhost:80";
+              "*.anotherone.com" = "http://localhost:80";
+            };
+          };
+        };
+      }));
+
+      default = { };
+    };
+  };
+
+  config = {
+    systemd.services =
```

Comment:
Is [this](https://github.com/cloudflare/cloudflared/blob/d6d65d183b5d215e9c383e1e3d5d0f8a6b21267e/cmd/cloudflared/linux_service.go#L54-L94) the basis for the unit?

Would there be any value in adding a `target` unit?